atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/pool/HostnamesByIpSupplier.java

```diff
@@ -43,20 +45,27 @@ public final class HostnamesByIpSupplier implements Supplier<Map<String, String>
     private static final String HOSTNAME_COLUMN = "hostname";
     private static final String IP_COLUMN = "ip";
 
-    private final Supplier<PoolingContainer<CassandraClient>> randomGoodHostSupplier;
+    private final Supplier<List<PoolingContainer<CassandraClient>>> hosts;
 
-    public HostnamesByIpSupplier(Supplier<PoolingContainer<CassandraClient>> randomGoodHostSupplier) {
-        this.randomGoodHostSupplier = randomGoodHostSupplier;
+    public HostnamesByIpSupplier(Supplier<List<PoolingContainer<CassandraClient>>> hosts) {
+        this.hosts = hosts;
     }
 
     @Override
     public Map<String, String> get() {
-        try {
-            return randomGoodHostSupplier.get().runWithPooledResource(getHostnamesByIp());
-        } catch (Exception e) {
-            log.warn("Could not get hostnames by ip from Cassandra", e);
-            return ImmutableMap.of();
-        }
+        return hosts.get().stream()
+                .map(container -> {
+                    try {
+                        return Optional.of(container.runWithPooledResource(getHostnamesByIp()));
+                    } catch (Exception e) {
+                        log.warn("Could not get hostnames by ip from Cassandra", e);
+                        return Optional.<Map<String, String>>empty();
+                    }
+                })
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .findFirst()
```

Comment:
If we're grabbing the first result, should we filter out any empty responses to grab the first non-empty view of `hostnamesByIp`?
```suggestion
                .map(Optional::get)
                .filter(hostnamesByIp -> !hostnamesByIp.isEmpty())
                .findFirst()
```

Reply:
I did this originally, but the (subtle) case that an empty map is a valid response, in the case that they mapping keyspace doesn't exist, table doesn't exist, or there are no rows in the map.

Reply:
When does the mapping keyspace or table not exist for a given node (e.g. is it only on bootstrap)? Aside from duration, why wouldn't we want to continue querying nodes to try and find a non-empty view from other nodes?

Related to the duration & 60 second timeout mentioned above, it feels like we might want to have something managing cluster state in background thread and possibly `Refreshable` mechanisms for updating & subscribing suppliers

Reply:
So the keyspace/table are an optional dependency, and are only needed in the case that AtlasDb is ran in a split environment (i.e. in two separate VPCs). As a result, we can expect that sometimes these tables do not exist, and in those cases, the IP addresses should be sufficient for communicating with. 

As for querying other nodes for non-empty views, the read itself is done at quorum, so there should be no difference in querying the 1 or 2 nodes, as that is what we're effectively doing under-the-hood.